src/active_query.rs

```diff
@@ -117,7 +118,7 @@ impl ActiveQuery {
             changed_at: self.changed_at,
             origin,
             durability: self.durability,
-            tracked_struct_ids: self.tracked_struct_ids.clone(),
+            tracked_struct_ids: self.tracked_struct_ids,
```

Comment:
There's one remaining place where `QueryRevisions` are deep cloned but I'm not sure if it justifies wrapping `tracked_struct_ids` in an `Arc`.

It might be cheaper to wrap `QueryRevisions` in an `Arc` instead (and remove the Arc around `QueryEdges``)?